spinalcordtoolbox/scripts/sct_detect_compression.py

```diff
@@ -0,0 +1,309 @@
+                                                                metrics_agg_merged[key[0] + 2,]['MEAN(orientation)']) +
+                                                            abs(metrics_agg_merged[key[0] - 2,]['MEAN(orientation)'] -
+                                                                metrics_agg_merged[key[0] - 3,]['MEAN(orientation)']) +
+                                                            abs(metrics_agg_merged[key[0] + 2,]['MEAN(orientation)'] -
+                                                                metrics_agg_merged[key[0] + 3,]['MEAN(orientation)']))
+            except Exception as e:
+                # TODO: the warning below is raise mainly in cases of no SC segmentation, for example above the C1 level
+                #  --> consider processing only slices with SC segmentation
+                printv(f"WARNING: Failed to compute torsion for slice {key[0]}. Use `-v 2` flag to see details",
+                       verbose, type='warning')
+                if verbose == 2:
+                    printv(f"\t{e}", verbose)
+                metrics_agg_merged[key]['Torsion'] = None
+        else:
+            metrics_agg_merged[key]['Torsion'] = None
+
+    return metrics_agg_merged
+
+
+def process_compression(metrics_agg_merged, disc_slices):
+    """
+    Process shape metrics for axial slices at the level of each disc to predict the compression probability.
+    :param metrics_agg_merged: dictionary with aggregated metrics
+    :param disc_slices: dictionary with disc labels as keys and corresponding axial slice numbers as values
+    :return: dataframe with compression probability
+    """
+    compression_df = pd.DataFrame()
+    # Loop across discs
+    for disc, slice in disc_slices.items():
+    # for disc, slices in disc_slices.items():
+    #     # Add one slice above and below to process multiple slices around the disc to compensate for potential
+    #     # disc label shift in the superior-inferior direction
+    #     slices = [slices - 1, slices, slices + 1]
+    #     for slice in slices:
+        if disc in SUPPORTED_DISCS:
+            # Note: [slice,] is used to convert int to tuple
+            cr = metrics_agg_merged[slice,]['MEAN(compression_ratio)'] * 100    # to convert to %
+            csa = metrics_agg_merged[slice,]['MEAN(area)']
+            solidity = metrics_agg_merged[slice,]['MEAN(solidity)'] * 100     # to convert to %
+            torsion = metrics_agg_merged[slice,]['Torsion']
+            # Compute compression probability
+            probability = predict_compression_probability(cr, csa, solidity, torsion, disc)
+            compression_category = 'high' if probability > CUT_OFF_HIGH else \
+                'moderate' if probability > CUT_OFF_MODERATE else 'low'
+            compression_df = pd.concat([compression_df,
+                                        pd.DataFrame([{'Disc': disc, 'Axial slice #': slice,
```

Comment:
We might reconsider the name of this column, `Axial slice #`. For example, `sct_process_segmentation` outputs `Slice (I->S)` column.